harvester_e2e_tests/fixtures/terraform.py

```diff
@@ -223,7 +223,9 @@ def tf_rancher(rancher_api_client, tf_script_dir, tf_provider_rancher_ver, tf_ex
 @pytest.fixture(scope="session")
 def tf_resource(tf_provider_version):
     converter = Path("./terraform_test_artifacts/json2hcl")
-    return TerraformResource.for_version(tf_provider_version)(converter)
+    # update `0.0.0-dev` to get newest class
+    version = "8.8.99" if tf_provider_version == '0.0.0-dev' else tf_provider_version
```

Comment:
qu: I'm sort of confused with the setting of `8.8.99`? is that to allign with something in:
https://github.com/harvester/tests/blob/586d6a718573a718f53ee3e7674fb0c647bf373f/apiclient/harvester_api/api.py#L38-L55

I was expecting this to read maybe something like:
```python
version = None if tf_provider_version == '0.0.0-dev' else tf_provider_version
```

:sweat_smile:  - but maybe I'm missing something...

Don't want this to be blocking, because other than this qu, this looks good :smille:

Reply:
yeah it is align to the version in `api.py`, you can check the explanation in https://github.com/harvester/tests/issues/1465#issuecomment-2310612259, the version in `tf_resource` will be used to get the correct implementation of `TerraformResource`, it should be a valid version (type: str), we will use it to determine which `TerraformResource` class we can use.

Classes of `TerraformResource` is implemented to align with the Provider version, for example, *Terraform Harvester Provider* implemented `cloudinit_secret` in `v0.6.3`[^1], so we implement `TerraformResource_063` to suppor the new implemented feature and mark it will `support_to = "0.6.3"`.

[^1]: https://github.com/harvester/terraform-provider-harvester/releases/tag/v0.6.3